### DISCUSSION TITLE
How can I get live output from a kitty window (something like tail -f @scrollback)? Alternatively, how to redirect STDOUT to another window?

### DISCUSSION THREAD

**Discussion by visrosa** [kovidgoyal/kitty#7619] [Q&A]
This is probably related to #5713, but instead of saving every output, I want to show it in another window. I can get the scrollback buffer, I can get and send text, but I can't auto-update it.

I was able to get something close to what I want snooping on /dev/pts/x with strace¹, but I assume I can do this with only kitty.
 
In case it's relevant, usecase is I'm trying to emulate Warp terminal's fixed prompt. I have one window where I input commands, and another window where I want the output to scroll by.

 [1] Thanks to github.com/lemonsqueeze/ttylogger

**Comment by kovidgoyal** [comment]
I dont know of anyway you can do that in kitty. kitty does not have any facility for notifying anything on tty IO, that's quite a performance hit. One could of course add code to do so, but I would only accept such code if it had *zero* performance impact when unused.

**Comment by visrosa** [comment]
That's too bad but to be expected. ttylogger also tanked performance.

What about redirecting all stdout? I did play around with adding ` | kitten @ send-text --stdin` to the end of every command, but it gets messy fast. 

Tmux can do it with pipe panes, apparently, but I don't really want to add another layer of complexity to my setup, specially since I managed to replicate tmux popups with kitty.

Update: I cheated a little and made a fake prompt that instead of executing the command just sends it to the window followed by enter. Works like a charm for what I wanted. Skeleton fish function:

```fish
function fake_prompt
	set -l cmd (commandline -co) (commandline -ct)
	kitten @ send-text -m 'title:Output' $cmd
	kitten @ send-key  -m 'title:Output' enter
	commandline -r ''
end
```

**Comment by kovidgoyal** [comment]
As I said you are welcome to contribute a PR adding it, as long as it
has no performance impact when not active. You can base it on the
existing infrastructure used by --dump-bytes
